### PR TITLE
Add MessageTemplate api to v4

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Basic add of MessageTemplate entity to apiv4  & use if from core form

Before
----------------------------------------
Not there

After
----------------------------------------
Ta-da

Technical Details
----------------------------------------
I'm looking to cleanup this form to use the apiv4 (I'd rather got the extra step & switch it to an afform but
that seems like too big a leap).

This switches the loading to use apiv4. Note that
1) I decided that it doesn't make sense to setCheckPermissions = FALSE - I think the form should not
be availble to non-permissioned users (& perhaps a hook might like to play a role here).

2) I removed the inheritence from the parent  which seemed to do 3 things
 - added admin.css - none of the classes seemed to apply
 - added iconpicker - didn't seem to apply
 - loaded the defaults - which this change does on the form more succinctly

Comments
----------------------------------------

### CODE REVIEW

**Comment by seamuslee001** [comment]
@eileenmcnaughton i think that needs to be a subdirectory up i.e. in Civi/Api4

**Comment by eileenmcnaughton** [comment]
opps you are right!

File: Civi/Api4/MsgTemplate.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC https://civicrm.org/licensing
+ */
```
Comment (colemanw) [inline Civi/Api4/MsgTemplate.php:17]: RIP `$Id$` we miss you.

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
@colemanw can you recheck this - I altered the form to use apiv4 as well & added some notes to the PR template & I'd like your feedback as apiv4 calls in core are fairly new

In the process of this I would note that I could also have used apiv3 & I can now see that the EntityFormTrait would have worked too (FYI  ) but I was kind of trying to get familiar with increasing v4 usage in core

@mattwire we should possibly switch the EntityFormTrait to use apiv4. I think we could just switch
```
    $metadata = civicrm_api3($this->getDefaultEntity(), 'getfields', ['action' => 'create']);
```
to 
```
    $metadata = civicrm_api4($this->getDefaultEntity(), 'getfields', ['action' => 'create']);
```
But we'd need some noisy fallback to apiv3 where there isn't a v4 entity

**Comment by colemanw** [comment]
Yes it all still looks good.

**Pull request merged**